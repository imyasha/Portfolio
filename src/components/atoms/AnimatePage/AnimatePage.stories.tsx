import { AnimatePage, AnimatePageProps } from './AnimatePage';

export default {
	title: 'atoms/AnimatePage',
	component: AnimatePage,
};

export const AnimatePageStory = (args: AnimatePageProps) => <AnimatePage {...args} />;

AnimatePageStory.storyName = 'AnimatePage';
AnimatePageStory.args = {};
