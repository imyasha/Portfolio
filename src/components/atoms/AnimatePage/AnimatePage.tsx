import { motion } from 'framer-motion';
import { FC, ReactNode } from 'react';

interface AnimatePageProps {
	children: ReactNode;
}

const AnimatePage: FC<AnimatePageProps> = ({ children }) => {
	const variants = {
		hidden: { opacity: 0, x: 0, y: 100 },
		enter: { opacity: 1, x: 0, y: 0 },
		exit: { opacity: 0, x: 0, y: -100 },
	};

	return (
		<motion.div
			initial="hidden"
			animate="enter"
			exit="exit"
			variants={variants}
		>
			{children}
		</motion.div>
	);
};

export { AnimatePage };
