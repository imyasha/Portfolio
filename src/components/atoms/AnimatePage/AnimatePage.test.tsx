import { AnimatePage } from '../AnimatePage';
import { render } from '@testing-library/react';

describe('AnimatePage', () => {
	it('renders correctly', () => {
		const { container } = render(<AnimatePage>Page Content</AnimatePage>);
		expect(container).toMatchSnapshot();
	});
});
