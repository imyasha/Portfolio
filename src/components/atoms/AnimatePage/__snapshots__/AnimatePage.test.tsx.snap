// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimatePage renders correctly 1`] = `
<div>
  <div
    style="opacity: 0; transform: translateX(0px) translateY(100px) translateZ(0);"
  >
    Page Content
  </div>
</div>
`;
